IPython/core/ultratb.py

```diff
@@ -336,13 +336,22 @@ def _format_exception_only(
                     )
                 )
                 if textline == "":
-                    # sep 2025:
-                    # textline = py3compat.cast_unicode(value.text, "utf-8")
-                    if value.text is None:
+                    # Gracefully coerce SyntaxError.text into a printable str.
+                    value_text = getattr(value, "text", None)
+                    if value_text is None:
                         textline = ""
                     else:
-                        assert isinstance(value.text, str)
-                        textline = value.text
+                        try:
+                            # Handle bytes by decoding to utf-8, fallback to str() for other types
+                            if isinstance(value_text, bytes):
+                                textline = value_text.decode("utf-8", "replace")
+                            else:
+                                textline = str(value_text)
+                        except Exception:
+                            # As a last resort, fall back to ``str`` for exotic objects.
+                            textline = str(value_text)
```

Comment:
SO... if `textline = str(value_text)` fails on line 349, you do `textline = str(value_text)` ... Do you see the problem ?